src/ophyd_async/tango/demo/_counter.py

```diff
@@ -19,7 +19,7 @@ class TangoCounter(TangoReadable):
     counts: SignalR[int]
     sample_time: SignalRW[float]
     start: SignalX
-    _reset: SignalX
+    reset_: SignalX
```

Comment:
@burkeds I've made it a trailing underscore rather than leading underscore to fit better with https://peps.python.org/pep-0008/#function-and-method-arguments

Reply:
That's fine with me. When signals are automatically created, we just have to make sure that this underscore placement is treated correctly.